internal/venice-common/src/main/java/com/linkedin/venice/hadoop/schema/HDFSSchemaSource.java

```diff
@@ -173,6 +223,31 @@ public Map<Integer, Schema> fetchValueSchemas() throws IOException {
     return mapping;
   }
 
+  @Override
+  public Schema fetchKeySchema() throws IOException {
+    FileStatus[] fileStatus = fs.listStatus(keySchemaDir);
```

Comment:
Should we short circuit and throw a more explicit exception in the case when `keySchemaDir ` is null and includeKeySchema is `false`?